pkg/cloud/services/asg/autoscalinggroup.go

```diff
@@ -30,12 +30,14 @@ import (
 	"sigs.k8s.io/cluster-api-provider-aws/pkg/record"
 )
 
-// SDKToAutoScalingGroup converts an AWS EC2 SDK AugoScalingGroup to the CAPA AugoScalingGroup type.
-// SDKToAugoScalingGroup populates all AugoScalingGroup fields
+// SDKToAutoScalingGroup converts an AWS EC2 SDK AutoScalingGroup to the CAPA AutoScalingGroup type.
 func (s *Service) SDKToAutoScalingGroup(v *autoscaling.Group) (*expinfrav1.AutoScalingGroup, error) {
 	i := &expinfrav1.AutoScalingGroup{
-		ID:   aws.StringValue(v.AutoScalingGroupARN),
-		Name: aws.StringValue(v.AutoScalingGroupName),
+		ID:              aws.StringValue(v.AutoScalingGroupARN),
+		Name:            aws.StringValue(v.AutoScalingGroupName),
+		DesiredCapacity: int32(aws.Int64Value(v.DesiredCapacity)),
```

Comment:
Interesting 🤔 Just wondering why there needed to be a lot of flipping back and forth between int32 and int64?

Reply:
Replica counts in Kubernetes are int32, the fields for amazon are *int64, so if we want to use the MachinePool spec.replicas as the desired count for the ASG, we need to convert. :\

Reply:
Got it, I didn't see that Replica counts in Kubernetes were int32, I only saw that AWS's was *int64 😄 👍